### PR TITLE
Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Harrynoqs**
Write integration tests with Capybara gem.
Secure app from n+1 problems.

218503420-36276066-ae71-450f-b258-002ae2fff869
218503414-d1588002-8d5b-498d-8f1f-9441e832d7b4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,121 +23,148 @@
...
+```
+ -->
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- AUTHORS -->
 
 ## 👥 Authors <a name="authors"></a>
 
+> Mention all of the collaborators of this project.
```
Comment (addod19) [inline README.md]: - [ ] Kindly remove this text, it's a placeholder text guiding us how to make our README file professional

File: README.md [hunk 2]
```diff
@@ -18,121 +23,148 @@
...
   - GitHub: [@Harrynoqs](https://github.com/Harrynoqs)
   - Twitter: [@Harrynoqs](https://twitter.com/noqsnoq1)
   - LinkedIn: [harrisonw-njuguna](https://linkedin.com/in/harrisonw-njuguna)
 
-
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 ## 🔭 Future Features <a name="future-features"></a>
 
+> Describe features you will add to the project.
```
Comment (addod19) [inline README.md]: - [ ] Kindly remove this

File: README.md [hunk 3]
```diff
@@ -148,21 +180,44 @@ Feel free to check the [issues page](../../issues/).
 
 ## ⭐️ Show your support <a name="support"></a>
 
-## 🙏 Acknowledgments <a name="acknowledgements"></a>
+> Write a message to encourage readers to support your project
+
+If you like this project...
```
Comment (addod19) [inline README.md]: - [ ] Kindly complete the text. If someone likes this project what should the person do?

File: README.md [hunk 4]
```diff
@@ -148,21 +180,44 @@ Feel free to check the [issues page](../../issues/).
...
 
-I would like to thank `Mike Dane` of `Giraffe Academy` and `freeCodeCamp.org` for wonderful tutorials. Big shout out to `Microverse` for an awesome curriculum. Hats off to my coding partners and my peers for a constant support. Without all of you, this project wouldn't be 
+<!-- ACKNOWLEDGEMENTS -->
 
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
+> Give credit to everyone who inspired your codebase.
 
-If you like this project kindly add ⭐️ to it.
+I would like to thank...
```
Comment (addod19) [inline README.md]: - [ ] Let us kindly complete the acknowledgement section

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @Harrynoqs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] All test cases run successfully :rocket: 
- [ ] Correct Git-flow followed :+1: 
- [ ] User and Post functionalities tested :1st_place_medal: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] There are a couple of file duplications that we need to remove, for example; `development.rb` is already in the environments directory and its also present in the root directory, also, `database.yml` which is also in the config directory finds itself in the root directory. This and many other files, let us kindly check them and remove them accordingly


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @Harrynoqs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passed ✔️
- Readme file professional ✔️ 

## Optional suggestions
[OPTIONAL] I would advise you to make all your commits/changes before submitting the project so that it would not cause confusion. I said this because I noticed there were changes you were making while I was reviewing it and it almost made me require changes meanwhile, you were still working on it.
[OPTIONAL] You must have wanted to write Future Features but you wrote key features and it makes key features duplicated and this is above the authors section.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**